src/uzfs_mgmt.c

```diff
@@ -539,6 +538,18 @@ update_zvol_property(zvol_state_t *zv, nvlist_t *nvprops)
 				} else if (prop == ZFS_PROP_WORKERS) {
 					zv->zvol_workers = (uint8_t)strtol(val,
 					    NULL, 10);
+				} else if (prop == ZFS_PROP_ZVOL_READONLY) {
+					mutex_enter(&zv->conf_mtx);
```

Comment:
how about moving this to a function which can be called from cstor as well

Reply:
done